### PR TITLE
Make kernel-install create entry directory

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: man/kernel-install.xml [hunk 1]
```diff
@@ -126,14 +126,23 @@
...
 
     </variablelist>
+  </refsect1>
 
+  <refsect1>
+    <title>The <varname>$BOOT</varname> partition</title>
+    <para>The partition where the kernels and <ulink url="https://systemd.io/BOOT_LOADER_SPECIFICATION">Boot
+    Loader Specification</ulink> snippets are located is called <varname>$BOOT</varname>.
+    <command>kernel-install</command> determines the location of this partition by checking
+    <filename>/efi/</filename>, <filename>/boot/</filename>, and <filename>/boot/efi</filename>
```
Comment (poettering) [inline man/kernel-install.xml:143]: we should probably change the order here, and use /boot/ first if the dirs exist there, and /efi then

Comment (poettering) [reply man/kernel-install.xml:143]: (but that might be stuff for another commit/PR)

File: src/boot/bootctl.c [hunk 2]
```diff
@@ -519,7 +519,7 @@ static int copy_file_with_version_check(const char *from, const char *to, bool f
 static int mkdir_one(const char *prefix, const char *suffix) {
         char *p;
 
-        p = strjoina(prefix, "/", suffix);
+        p = path_join(prefix, suffix);
```
Comment (poettering) [inline src/boot/bootctl.c:522]: this changes allocation from stack to heap. Needs a `_cleanup_free_` decorator on `char *p` hence.

File: man/kernel-install.xml (hunk 1 above)
Comment (keszybz) [reply man/kernel-install.xml:143]: Ack.

**Comment by keszybz** [comment]
Updated to fix the memleak. No other changes.

**Pull request merged**